SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -2610,7 +2611,7 @@ void CEulerSolver::CommonPreprocessing(CGeometry *geometry, CSolver **solver_con
 
   if(!ReducerStrategy && !Output) {
     LinSysRes.SetValZero();
-    if (implicit && !disc_adjoint) Jacobian.SetValZero();
+    if (implicit) Jacobian.SetValZero();
```

Comment:
The changes in residuals of the AD tests are partly due to this change, we were not clearing the matrix in AD mode, which means when multigrid is on we would add on top of the matrix at least once (in practice nPreSmooth + 1 + nPostSmooth) which I think explains why all the residuals I looked at changed for the better (since doubling the matrix halves the correction, kinda).

Reply:
This is the improvement in convergence rate for the first case that failed Arina2K:
![image](https://user-images.githubusercontent.com/38071223/80867927-ebf26f00-8c8e-11ea-9b5b-b0ac40ff75b1.png)
The picture is similar for the other discadj cases with MG, at convergence the sensitivities and adjoints are "the same". I've updated the regressions.